boards/DIM/Src/App/states/App_DriveState.c

```diff
@@ -190,3 +199,18 @@ const struct State *App_GetDriveState(void)
 
     return &drive_state;
 }
+
+bool App_DriveState_HasBmsFault(const struct DimCanRxInterface *can_rx)
+{
+    uint8_t test = App_CanRx_BMS_STATE_MACHINE_GetSignal_STATE(can_rx) == CANMSGS_BMS_STATE_MACHINE_STATE_FAULT_CHOICE;
+    return test;
+}
+
+bool App_DriveState_HasBmsWarning(const struct DimCanRxInterface *can_rx)
+{
+    return App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1_HZ(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1_KHZ(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANRX(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANTX(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_WATCHDOG_TIMEOUT(can_rx);
+}
```

Comment:
I agree it's not great, especially once other boards use a similar format. What's not demonstrated here however is that this would allow us to pick and choose which warnings trigger the LED on the DIM, such that warnings irrelevant to the driver can be ignored.

Reply:
Good point, and yeah actually there probably will be aperiodic alerts that don't indicate a warning, such as startup messages.